### PR TITLE
Introduced function to create byte buffers from the output of a byte buffer reader.

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

**Comment by nicolasnoble** [comment]
test this please

**Comment by dgquintas** [comment]
Travis is still broken for ruby somehow.

**Comment by soltanmm** [comment]
cc #2111

File: tools/run_tests/build_ruby.sh [hunk 1]
```diff
@@ -37,4 +37,4 @@ export CONFIG=${CONFIG:-opt}
 cd $(dirname $0)/../../src/ruby
 
 bundle install
-rake compile:grpc
+rake compile:grpc || cat tmp/*/*/*/mkmf.log
```
Comment (ctiller) [inline tools/run_tests/build_ruby.sh]: Did you intend to include this?

Comment (dgquintas) [inline tools/run_tests/build_ruby.sh]: Nope, removing. That was a test while trying to solve the Ruby issue with Nico.

**Comment by dgquintas** [comment]
test this please

**Comment by dgquintas** [comment]
OMG GREEN. (Travis).

This PR also contains the changes needed by Ruby to link zlib (whose absence were making Ruby fail for any compression-related changes).

**Pull request merged**